core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpLineageDispatcher.scala

```diff
@@ -87,4 +145,33 @@ class HttpLineageDispatcher(splineServerRESTEndpointBaseURL: String, http: BaseH
       case _ => Unit
     }
   }
+
+  private def isServerCompressionReady(): Boolean = {
+    val tryStatus = Try(baseHttp(statusUrl)
+      .method("GET")
+      .asString
+      .throwError
+      .body)
+
+    tryStatus
+      .toOption
+      .map(extractCompressionFlag(_))
+      .getOrElse(false)
+  }
+
+  private def extractCompressionFlag(json: String): Boolean = {
```

Comment:
Can we use JSON commons for this?

Reply:
I would like this to work without need for specific format of the incoming JSON, that would make it forward compatible when we add more fields.

Reply:
`JsonSerDe` can do it as well, and it's a one-liner.
See a usage example in a test here - https://github.com/AbsaOSS/commons/blob/master/src/test/scala/za/co/absa/commons/json/JsonSerDeSpec.scala#L63

Reply:
```scala
val isCompressionSupported = body.fromJson[Map[String, Boolean]].getOrElse("requestDecompression", false)
```